pkg/scheduler/cache/cache.go

```diff
@@ -176,6 +183,123 @@ type imageState struct {
 	nodes sets.Set[string]
 }
 
+// MemberMatcher represents a matcher for a child node (exact for regex)
+type MemberMatcher struct {
+	Exact string // Exact string match
+	Regex string // Regex match
+}
+
+// Matches checks if the node name matches the MemberMatcher
+func (mm *MemberMatcher) Matches(name string) bool {
+	if mm.Exact != "" {
+		result := mm.Exact == name
+		klog.V(4).Infof("Matching node %s with exact %s: %v", name, mm.Exact, result)
+		return result
+	}
+	if mm.Regex != "" {
+		r, err := regexp.Compile(mm.Regex)
+		if err != nil {
+			klog.Errorf("Failed to compile regex %s: %v", mm.Regex, err)
+			return false
+		}
+		result := r.MatchString(name)
+		klog.V(4).Infof("Matching node %s with regex %s: %v", name, mm.Regex, result)
+		return result
+	}
+	return false
+}
+
+// NewExactMatcher creates a new MemberMatcher for exact matching
+func NewExactMatcher(exact string) *MemberMatcher {
+	return &MemberMatcher{Exact: exact}
+}
+
+// NewRegexMatcher creates a new MemberMatcher for regex matching
+func NewRegexMatcher(regex string) (*MemberMatcher, error) {
+	_, err := regexp.Compile(regex)
+	if err != nil {
+		klog.Errorf("Failed to compile regex %s: %v", regex, err)
+		return nil, err
+	}
+	return &MemberMatcher{Regex: regex}, nil
+}
+
+// HyperNodeTreeNode reprensent a single hypernode in the tree
+type HyperNodeTreeNode struct {
+	Name           string
+	Tier           int
+	Parent         *HyperNodeTreeNode
+	Members        sets.Set[*HyperNodeTreeNode]
+	MemberMatchers sets.Set[*MemberMatcher]
```

Comment:
Can we group together Members/NodeMatchers and MemberMatchers/NodeMatchers? Too many unused fields